src/lsdb/loaders/dataframe/from_dataframe_utils.py

```diff
@@ -31,6 +33,8 @@ def _generate_dask_dataframe(
     divisions = get_pixels_divisions(pixels)
     ddf = dd.from_delayed(delayed_dfs, meta=schema, divisions=divisions)
     ddf = ddf if isinstance(ddf, dd.DataFrame) else ddf.to_frame()
+    ddf = NestedFrame.from_dask_dataframe(ddf)
```

Comment:
This chunk of type checking/casting looks like it could be streamlined.

Reply:
It can! NestedFrame has a method that gets around the type checking needed for dask